File: shanoir-ng-front/src/app/preclinical/anesthetics/examination_anesthetic/shared/examinationAnesthetic.service.ts
/**
 * Shanoir NG - Import, manage and share neuroimaging data
 * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
 * Contact us on https://project.inria.fr/shanoir/
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
 */

import { Injectable } from '@angular/core';
import { EntityService } from '../../../../shared/components/entity/entity.abstract.service';
import { ExaminationAnesthetic } from './examinationAnesthetic.model';
import * as PreclinicalUtils from '../../../utils/preclinical.utils';
import { HttpClient } from '@angular/common/http';

@Injectable()
export class ExaminationAnestheticService extends EntityService<ExaminationAnesthetic>{
    API_URL = PreclinicalUtils.PRECLINICAL_API_EXAMINATION_URL;   
    
    constructor(protected http: HttpClient) {
        super(http)
    }
    
    getEntityInstance() { return new ExaminationAnesthetic(); }      
    
    getExaminationAnesthetics(examination_id:number): Promise<ExaminationAnesthetic[]>{
        const url = `${PreclinicalUtils.PRECLINICAL_API_EXAMINATION_URL}/${examination_id}/${PreclinicalUtils.PRECLINICAL_ANESTHETIC}${PreclinicalUtils.PRECLINICAL_ALL_URL}`;
        return this.http.get<ExaminationAnesthetic[]>(url)
        .toPromise()
        .then(entities => entities.map((entity) => this.toRealObject(entity)));
    }

    
    
    getExaminationAnesthetic(examination_id:number,eaid: number): Promise<ExaminationAnesthetic> {
        const url = `${PreclinicalUtils.PRECLINICAL_API_EXAMINATION_URL}/${examination_id}/${PreclinicalUtils.PRECLINICAL_ANESTHETIC}/${eaid}`;
        return this.http.get<ExaminationAnesthetic>(url)
        .toPromise()
        .then((entity) => this.toRealObject(entity));
    }
    
    getAllExaminationForAnesthetic(aid: number): Promise<ExaminationAnesthetic[]> {
        const url = `${PreclinicalUtils.PRECLINICAL_API_EXAMINATION_URL}${PreclinicalUtils.PRECLINICAL_ALL_URL}/${PreclinicalUtils.PRECLINICAL_ANESTHETIC}/${aid}`;
        return this.http.get<ExaminationAnesthetic[]>(url)
                .toPromise()
                .then(response => response)
                .catch((error) => {
                    console.error('Error while getting ExaminationAnesthetic for an Anesthetic', error);
                    return Promise.reject(error.message || error);
                });
    }
    
    updateAnesthetic(examination_id:number, examAnesthetic: ExaminationAnesthetic): Promise<ExaminationAnesthetic> {
        const url = `${PreclinicalUtils.PRECLINICAL_API_EXAMINATION_URL}/${examination_id}/${PreclinicalUtils.PRECLINICAL_ANESTHETIC}/${examAnesthetic.internal_id}`;
        return this.http
            .put<ExaminationAnesthetic>(url, JSON.stringify(examAnesthetic)) 
            .toPromise();
        }
    
    createAnesthetic(examination_id:number, examAnesthetic: ExaminationAnesthetic): Promise<ExaminationAnesthetic> {
        const url = `${PreclinicalUtils.PRECLINICAL_API_EXAMINATION_URL}/${examination_id}/${PreclinicalUtils.PRECLINICAL_ANESTHETIC}`;
            return this.http
            .post<ExaminationAnesthetic>(url, JSON.stringify(examAnesthetic))
            .toPromise();
        }

    deleteAnesthetic(examAnesthetic: ExaminationAnesthetic): Promise<void> {
        const url = `${PreclinicalUtils.PRECLINICAL_API_EXAMINATION_URL}/${examAnesthetic.examination_id}/${PreclinicalUtils.PRECLINICAL_ANESTHETIC}/${examAnesthetic.id}`;
        return this.http.delete<void>(url)
            .toPromise()
    }
    
}

File: shanoir-ng-preclinical/src/main/java/org/shanoir/ng/preclinical/anesthetics/examination_anesthetics/ExaminationAnestheticService.java
/**
 * Shanoir NG - Import, manage and share neuroimaging data
 * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
 * Contact us on https://project.inria.fr/shanoir/
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
 */

package org.shanoir.ng.preclinical.anesthetics.examination_anesthetics;

import java.util.List;

import org.shanoir.ng.preclinical.anesthetics.anesthetic.Anesthetic;
import org.shanoir.ng.shared.exception.ShanoirException;
import org.shanoir.ng.shared.validation.UniqueCheckableService;

/**
 * Examination anesthetic service.
 *
 * @author sloury
 *
 */
public interface ExaminationAnestheticService extends UniqueCheckableService<ExaminationAnesthetic> {

	/**
	 * Delete an examination anesthetic
	 * 
	 * @param id
	 *            examination anesthetic id.
	 * @throws ShanoirException
	 */
	void deleteById(Long id) throws ShanoirException;

	/**
	 * Get all the examination anesthetics
	 * 
	 * @return a list of examination anesthetics
	 */
	List<ExaminationAnesthetic> findAll();

	/**
	 * Get all the examination anesthetics by examination id
	 * 
	 * @return a list of examination anesthetics
	 */
	List<ExaminationAnesthetic> findByExaminationId(Long examinationId);

	/**
	 * Find examination anesthetic by its id.
	 *
	 * @param id
	 *            examination anesthetic id.
	 * @return a examination anesthetic or null.
	 */
	ExaminationAnesthetic findById(Long id);

	/**
	 * Save an examination anesthetic
	 *
	 * @param examination
	 *            anesthetic examination anesthetic to create.
	 * @return created ExaminationAnesthetic.
	 * @throws ShanoirException
	 */
	ExaminationAnesthetic save(ExaminationAnesthetic examAnesthetic) throws ShanoirException;

	/**
	 * Update a examination anesthetic
	 *
	 * @param examination
	 *            anesthetic examination anesthetic to update.
	 * @return updated ExaminationAnesthetic.
	 * @throws ShanoirException
	 */
	ExaminationAnesthetic update(ExaminationAnesthetic examAnesthetic) throws ShanoirException;

	/**
	 * Get all the examination anesthetics by anesthetic
	 * 
	 * @return a list of examination anesthetics
	 */
	List<ExaminationAnesthetic> findByAnesthetic(Anesthetic anesthetic);

}
